code/src/components/Labyrinth.js

```diff
@@ -0,0 +1,223 @@
+
+// Reducers & Functions
+import { fetchLabyrinthData } from '../reducers/labyrinth';
+import { labyrinth } from '../reducers/labyrinth';
+
+// Components
+import { Button } from './Button';
+import { NameInput } from './NameInput';
+import { DirectionButtons } from './DirectionButtons';
+import { Header } from './Header';
+import { Footer } from './Footer';
+import { History } from './History';
+
+// Styling & Images
+import { InnerFlexWrapper, OuterFlexWrapper } from '../styling/GlobalStyles';
+import {
+  imgURL_start,
+  imgURL_0_0,
+  imgURL_0_1,
+  imgURL_0_2,
+  imgURL_0_3,
+  imgURL_1_0,
+  imgURL_1_1,
+  imgURL_1_3,
+} from '../styling/ImageSources';
+
+// ----------------------------------------------------------------
+
+export const Labyrinth = ({ setCurrentCoordinates }) => {
+  const dispatch = useDispatch();
+  const content = useSelector((store) => store.labyrinth.content);
+  const username = useSelector((store) => store.labyrinth.username);
+  const isLoading = useSelector((state) => state.ui.isLoading);
+  const history = useSelector((state) => state.labyrinth.history);
+
+  const [startButtonVisible, setStartButtonVisible] = useState(false);
+  const [nameInputVisible, setNameInputVisible] = useState(true);
+  const [historyVisible, setHistoryVisible] = useState(false);
+
+  // Check if username is stored in local storage
+  const currentUsername = localStorage.getItem('username')
+    ? localStorage.getItem('username')
+    : username;
+
+  // Restart game - reset all values and reload
+  const handleRestartButton = () => {
+    dispatch(labyrinth.actions.setUsername(''));
+    localStorage.setItem('labyrinth', '{}');
+    localStorage.setItem('username', '');
+    localStorage.setItem('history', '[]');
+    window.location.reload();
+  };
+
+  // Send current coordinates to parent component for styling
+  setCurrentCoordinates(content.coordinates);
+
+  return (
+    isLoading === false && (
+      <MainWrapper coordinates={content.coordinates}>
+        <Header />
+        <InnerWrapper>
+          {/* Removes the NameInput and start buttons when the journey has started */}
+          {content.coordinates === undefined && (
+            <>
+              {/* Name-input */}
+              {nameInputVisible && (
+                <NameInput
+                  setStartButtonVisible={setStartButtonVisible}
+                  setNameInputVisible={setNameInputVisible}
+                />
+              )}
+
+              {/* Start-button */}
+              {startButtonVisible && (
+                <InnerFlexWrapper>
+                  <IntroText>
+                    Welcome {currentUsername}, are you ready to start your
+                    journey?
+                  </IntroText>
+                  <Button
+                    action={() =>
+                      fetchLabyrinthData({
+                        url: 'https://wk16-backend.herokuapp.com/start',
+                        username: currentUsername,
```

Comment:
These are actually props for the thunk-function. As Damien pointed out we made the function so that using props is possible. And by doing so we avoid having to write two thunks that are identical to each other, ie. saving about 30 lines of code in the reducer. 😊